### PR TITLE
Added stripe element and event to actions

### PR DESCRIPTION
**Pull request by ivanvanderbyl**
This PR adds support for sending the `stripeElement` and event object as arguments to the event handlers, as well as adding new actions: 

- `error` which fires only when the change event fires and the card is invalid. This can be used in a similar way to `complete` for more complex error state handling, like disabling a form submit or show error messages outside the block.
- `complete` which only fires once the card input is complete. This allows for sending the stripeElement up to another form object or controller to handle complex nested payment forms. For example:

```hbs
{{#form-for changeset submit=(action "createCustomer") as |f|}}
  {{stripe-card complete=(mut changeset._stripeElement)}}
  {{f.text-field "billingEmail" placeholder="accounts@mycompany.com" label="Send receipts to:"}}
  {{f.submit "Begin Subscription"}}
{{/form-for}}
```

```js
createCustomer(changeset) {
  return changeset.validate().then(() => 
    if(changeset.get('isValid')) {
      stripe.createToken(changeset.get('_stripeElement')).then(({token}) => {
        set(changeset, 'cardToken', token);
      });
      ...
    }
  });
}

```

### CODE REVIEW

File: addon/components/stripe-element.js [hunk 1]
```diff
@@ -66,12 +66,18 @@ export default Component.extend({
...
     stripeElement.on('change',  (...args) => {
-      let { error } = args[0];
+      let { error, complete } = args[0];
       set(this, 'error', error);
-      this.sendAction('change', ...args);
+      this.sendAction('change', stripeElement, ...args);
+
+      if (complete === true) {
+        this.sendAction('complete', stripeElement);
+      }else if(error) {
```
Comment (JoshSmith) [inline addon/components/stripe-element.js]: `} else if (error) {`

Also, is it really `complete || error`?

**Comment by JoshSmith** [comment]
@ivanvanderbyl thanks for this PR!

Overall this solves some of the design issues with the blocked usage.

A few things I would love to see here:

- Some additions to the README that mention this usage
- Possibly some additions to the dummy app to also show off what kind of complex behavior can emerge
- Mentioning usage of `yarn` in the README
- If you have any thoughts around relevant tests

Although not relevant to the PR specifically, I also noticed we should probably have:

- eslint that runs on Travis, so we could've caught the issue with the `head` tag insertion, and the spacing comment I added

**Comment by JoshSmith** [comment]
I started in #7 getting coverage working, so it might be nice to wrap that up first if we can discuss some testing ideas (being that the `iframe` makes much of this difficult). With testing wrapped up, I'll feel much more comfortable about such large surface area changes to the actions/events.

File: addon/components/stripe-element.js (hunk 1 above)
Comment (ivanvanderbyl) [inline addon/components/stripe-element.js]: Yeah maybe it should just be the default `action`? Also, do you have an aversion to prefixing event based actions with `on-`?

**Comment by ivanvanderbyl** [comment]
Cool, yep I'll update the readme and address those other points once I get this feature shipped in our app.

With the block usage, the main hurdle design wise is lacking any way to specify where the Stripe Element is placed. Maybe we could yield a sub-component which just renders the placeholder div where it's needed, but then maybe that isn't a design concern for this addon. Wouldn't want to complicate it unnecessarily.

**Comment by ivanvanderbyl** [comment]
Testing the iframe could be tricky, but i have seen something which did this before, I think it was outside of the ember landscape though.

**Comment by JoshSmith** [comment]
@ivanvanderbyl awesome to hear! Are you in the Ember slack? Or you could join ours http://slack.codecorps.org

**Comment by JoshSmith** [comment]
Hey @ivanvanderbyl just wanted to check in on this with you.

**Comment by JoshSmith** [comment]
Hi @ivanvanderbyl I would still love to see about this getting in. Have you had any time to spend on this?

**Comment by JoshSmith** [comment]
Ping.

**Comment by JoshSmith** [comment]
Opened #17 as a follow-up to this since I'd like to get this merged.

Thanks again for the PR @ivanvanderbyl! 🙌

**Pull request merged**